rotkehlchen/tests/unit/decoders/test_hop.py

```diff
@@ -52,10 +85,57 @@ def test_hop_l2_deposit(database, ethereum_inquirer):
             notes='Bridge 0.2 ETH to Optimism at the same address via Hop protocol',
             counterparty=CPT_HOP,
             address=string_to_evm_address('0xb8901acB165ed027E32754E0FFe830802919727f'),
+            extra_data={'sent_amount': '0.2'},
         )]
     assert expected_events == events
 
 
+# @pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [['0xD8245043a3f339400dcfFFc7A9E6F22a264b933D']])
+def test_hop_l2_deposit_usdc(database, ethereum_inquirer, ethereum_accounts):
+    """Data taken from
+    https://etherscan.io/tx/0xd46640417a686b399b2f2a920b0c58a35095759365cbe7b795bddec34b8c5eee
+    """
+    tx_hash = deserialize_evm_tx_hash('0xac42ca2d88194c0ee219f6c71c98fe566667151a1dc235d17d993b6985342062')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp = TimestampMS(1710955643000)
+    gas_fee, bridge_amount = '0.005802319111323689', '3700'
+    expected_events = [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(gas_fee)),
+            location_label=ethereum_accounts[0],
+            notes=f'Burned {gas_fee} ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=363,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.BRIDGE,
+            asset=A_USDC,
+            balance=Balance(amount=FVal(bridge_amount)),
+            location_label=ethereum_accounts[0],
+            notes=f'Bridge {bridge_amount} USDC to Arbitrum One at the same address via Hop protocol',  # noqa: E501
```

Comment:
I believe we don't add to the same address. It is meaningful only if the address is different from the user address. So I would add that part only if target_address != origin address

Reply:
Updated